### PR TITLE
SDIT-873 Better validation and evidence creation

### PR DESCRIPTION
**Pull request by andymarke**
* Evidence creation
* Better validation
* Better tests around validation errors
* Minor refractor in the service to improve readability

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/CreateAdjudicationRequest.kt [hunk 1]
```diff
@@ -19,39 +19,42 @@ data class CreateAdjudicationRequest(
   @Schema(description = "Charges associated with this adjudication")
   val charges: List<ChargeToCreate>,
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/CreateAdjudicationRequest.kt:21]: moved this from Incident to top level which better represents the division between Incident and Adjudication

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/CreateAdjudicationRequest.kt [hunk 2]
```diff
@@ -19,39 +19,42 @@ data class CreateAdjudicationRequest(
...
-  @Schema(description = "Charges associated with this adjudication (business key)")
+  @Schema(description = "Charges associated with this adjudication (business key)", example = "1234567/1")
   val offenceId: String,
 )
 
 @JsonInclude(JsonInclude.Include.NON_NULL)
 data class IncidentToCreate(
-  @Schema(description = "Reporting staff member username", required = true)
+  @Schema(description = "Reporting staff member username", required = true, example = "JANE.BROOKES")
   val reportingStaffUsername: String,
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/CreateAdjudicationRequest.kt:37]: improve swagger examples

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/AdjudicationIncident.kt [hunk 3]
```diff
@@ -88,5 +88,5 @@ class AdjudicationIncident(
   override fun hashCode(): Int = javaClass.hashCode()
 }
 
-fun AdjudicationIncident.findAdjudication(adjudicationNumber: Long): AdjudicationIncidentParty =
-  this.parties.findAdjudication(adjudicationNumber)
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/AdjudicationIncident.kt:92]: slight improvement to readability on usage

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**